tests/event.test.ts

```diff
@@ -0,0 +1,130 @@
+    ["__/Link__", "__https://html-link__"],
+    ["__/Hangout__", "__https://meet-link__"],
+    ["__/Location__", "__Room 22__"],
+    ["__/Start Time__", "__08:00 AM__"],
+    ["__/End Time__", "__08:30 AM__"],
+
+  ]).test("Format event with format: %s", (format: string, expected: string) => {
+    expect(formatEvent(event, format, false)).toBe(expected);
+  });
+});
+
+describe("Test the effect of the Include Link setting", () => {
+  test("Test including event link", () => {
+    expect(
+      formatEvent(event, "__{summary}__", true)
+    ).toBe("__[Meditation retreat](https://html-link)__")
+  });
+
+  test("Test including event link does not change deprecated format", () => {
+    expect(
+      formatEvent(event, "__/Summary__", true)
+    ).toBe("__Meditation retreat__")
+  });
+});
+
+describe("Test date formatting", () => {
+  each([
+    ["", "2021-09-01T21:00:00+00:00", "09:00 PM"],
+    ["hh:mm", "2021-09-01T21:00:00+00:00", "09:00"],
+    ["HH:mm", "2021-09-01T21:00:00+00:00", "21:00"],
+    ["h:mm:ss a", "2021-09-01T21:00:00+00:00", "9:00:00 PM"],
+  ]).test("Test dates with formatting: %s",
+    (date_format: string, date: string, expected: string) => {
+      expect(
+        formatEvent(eventFactory(
+          {start: {dateTime: date}}), `{start:${date_format}}`, false)
+      ).toBe(expected)
+    });
+});
+
+describe("Test attendees formatting", () => {
+
+  const emailEvent = event;
+  const namesEvent = eventFactory({
+    attendees: [
+      {
+        email: "me@example.com",
+        displayName: "Alice Jane"
+      },
+      {
+        email: "other@example.com",
+        displayName: "Bob John"
+      }
+    ]
+  });
+
+  each([
+    ["", "me@example.com, other@example.com"],
+    ["NAME", "me@example.com, other@example.com"],
+    ["[[NAME]]", "[[me@example.com]], [[other@example.com]]"],
+    ["[NAME](mailto:NAME)", "[me@example.com](mailto:me@example.com), [other@example.com](mailto:other@example.com)"],
+  ]).test("Test attendees (only email) with formatting: %s",
+    (attendee_format: string, expected: string) => {
+      expect(
+        formatEvent(emailEvent, `{attendees:${attendee_format}}`, false)
+      ).toBe(expected)
+    });
+
+  each([
+    ["", "Alice Jane, Bob John"],
+    ["NAME", "Alice Jane, Bob John"],
+    ["[[NAME]]", "[[Alice Jane]], [[Bob John]]"],
+  ]).test("Test attendees (with display name) with formatting: %s",
+    (attendee_format: string, expected: string) => {
+      expect(
+        formatEvent(namesEvent, `{attendees:${attendee_format}}`, false)
+      ).toBe(expected)
+    });
+
+
+});
+
+describe("Test confLink formatting", () => {
+
+  const formattedEvent = (zoom: boolean, meet: boolean) => {
+    const overrides = zoom ? {location: "https://us02web.zoom.us/j/000"} : {};
+    let event = eventFactory(overrides);
+    if (!meet) {
+      delete event.hangoutLink;
+    }
+    return formatEvent(event, "{confLink}", false);
+  };
+
+  each`
```

Comment:
is this just a string next to `each`? this doesn't error?

Reply:
not a string, but a [Tagged Template Literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)!

Does not error, look at the actions ran in this PR, it expands the ASCII table to be one test per row.

Reply:
> Tagged, this calls the function "example" with the template as the first argument and substitution values as subsequent arguments

woah. was completely unaware of this!